jsonpickle/util.py

```diff
@@ -43,10 +43,19 @@ def is_type(obj):
     >>> is_type(Klass)
     True
     """
+    # use "isinstance" and not "is" to allow for metaclasses
     if PY3:
-        return type(obj) is type
+        return isinstance(obj, type)
     else:
-        return type(obj) is type or type(obj) is types.ClassType
+        return isinstance(obj, (type, types.ClassType))
+
+
+def has_method(obj, name, *args, **kwargs):
+    try:
+        getattr(obj, name)(*args, **kwargs)
+        return True
+    except:
+        return False
```

Comment:
This enum object is pretty interesting :-) poking at it a bit, I did notice that `Foo.A.__getnewargs__.__self__` exists, so maybe the `__self__` attribute can give us additional clues. I'm mentioning it just in case it can help us out in this situation. Excellent sleuthing.